### PR TITLE
Add Glossary CRUD Page

### PR DESCRIPTION
**Pull request by mjoyce91**
Adds the Glossary CRUD view and actions. Glossary Editor can currently be accessed by any user from the profile menu. Users can search by keyword and sort by first letter of the glossary term, which all takes place client-side. Draftjs is used to update the title and definition.

Placeholders:
- Nav links in Glossary Editor
- History link in cards
- ~~Updated on/by in cards~~

To-do in another PR:
- ~~Archive button~~ Added ✅

### CODE REVIEW

File: src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    let filteredTerms = filterByProps(searchText, ['title', 'definition'], glossaryItems);
+    filteredTerms = groupBy(filteredTerms, (term) => {
+      const first = term.title.substr(0, 1).toUpperCase();
+      // check if it's a letter
+      const firstIsAlpha = first.match(/^[a-zA-Z]*$/);
+      // if so, assign it to the '#' prop
+      if (firstIsAlpha) {
+        return first;
+      // else, assign it to its first letter
+      } return '#';
```
Comment (burgwyn) [inline src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx]: 🔢

File: src/Components/BidderPortfolio/TopNav/Navigation/NavigationItem/NavigationItem.jsx [hunk 2]
```diff
@@ -7,21 +7,27 @@ import { isCurrentParam } from '../../../../ProfileMenu/navigation';
...
-  numerator: PropTypes.number.isRequired,
-  denominator: PropTypes.number.isRequired,
+  numerator: PropTypes.number,
+  denominator: PropTypes.number,
   link: PropTypes.string.isRequired,
   location: ROUTER_LOCATION_OBJECT.isRequired,
 };
 
+NavigationItem.defaultProps = {
+  numerator: undefined,
```
[the remark this answers is not in the document]
Comment (mjoyce91) [reply src/Components/BidderPortfolio/TopNav/Navigation/NavigationItem/NavigationItem.jsx]: If it's undefined, we don't try to render anything related to it.

File: src/Components/GlossaryEditor/GlossaryEditorCard/GlossaryEditorCard.jsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            />
+          }
+          {
+            showEmptyWarning &&
+            <div>
+              <span className="usa-input-error-message" role="alert">Title or definition cannot be blank</span>
+            </div>
+          }
+        </div>
+        <div className="usa-grid-full glossary-editor-card-bottom">
```
[the remark this answers is not in the document]
Comment (mjoyce91) [reply src/Components/GlossaryEditor/GlossaryEditorCard/GlossaryEditorCard.jsx]: I'll have real data in the next PR

File: src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx (hunk 1 above)
Comment (mjoyce91) [reply src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx]: These comments should actually be flipped

File: src/Components/GlossaryEditor/GlossaryEditorCard/GlossaryEditorCard.jsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+              initialText={renderedDefinition}
+              onSubmitText={this.submitDefinition}
+              cancel={this.cancel}
+              onChangeText={this.updateDefinition}
+              draftJsProps={{ ariaDescribedBy: 'input-error-message' }}
+            />
+          }
+          {
+            showEmptyWarning &&
+            <div>
```
[the remark this answers is not in the document]
Comment (mjoyce91) [reply src/Components/GlossaryEditor/GlossaryEditorCard/GlossaryEditorCard.jsx]: Included in next PR

**Comment by mjoyce91** [comment]
@burgwyn - added more updates to this PR, which address a few of your comments.

**Comment by mjoyce91** [comment]
Upon further review, I have quite a few inline styles I need to abstract to classes. 😱

File: src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx (hunk 1 above)
Comment (mjoyce91) [reply src/Components/GlossaryEditor/GlossaryEditorPage/GlossaryEditorPage.jsx]: Updated

**Pull request merged**